share/spack/gitlab/cloud_pipelines/stacks/aws-isc/spack.yaml

```diff
@@ -0,0 +1,246 @@
+spack:
+  view: false
+  concretization: separately
+
+  concretizer:
+    reuse: false
+
+  config:
+    concretizer: clingo
+    install_tree:
+      root: /home/software/spack
+      padded_length: 384
+      projections:
+        all: '{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+
+  packages:
+    all:
+      providers:
+        blas:
+          - openblas
+        mkl:
+          - intel-oneapi-mkl
+        mpi:
+          - openmpi
+          - mpich
+      variants: +mpi
+    binutils:
+      variants: +ld +gold +headers +libiberty ~nls
+      version:
+        - 2.36.1
+    doxygen:
+      version:
+        - 1.8.20
+    elfutils:
+      variants: +bzip2 ~nls +xz
+    hdf5:
+      variants: +fortran +hl +shared
+    libfabric:
+      variants: fabrics=efa,tcp,udp,sockets,verbs,shm,mrail,rxd,rxm
+    libunwind:
+      variants: +pic +xz
+    mesa:
+      variants: ~llvm
+    mesa18:
+      variants: ~llvm
+    mpich:
+      variants: ~wrapperrpath netmod=ofi device=ch4
+    ncurses:
+      variants: +termlib
+    openblas:
+      variants: threads=openmp
+    openmpi:
+      variants: fabrics=ofi +legacylaunchers
+    openturns:
+      version: [1.18]
+    relion:
+      variants: ~mklfft
+#    texlive:
+#      version: [20210325]
+    trilinos:
+      variants: +amesos +amesos2 +anasazi +aztec +belos +boost +epetra +epetraext +ifpack +ifpack2 +intrepid +intrepid2 +isorropia +kokkos +ml +minitensor +muelu +nox +piro +phalanx +rol +rythmos +sacado +stk +shards +shylu +stokhos +stratimikos +teko +tempus +tpetra +trilinoscouplings +zoltan +zoltan2 +superlu-dist gotype=long_long
+    xz:
+      variants: +pic
+
+  definitions:
+
+  - compiler_specs:
+    - gcc@11.2
+    # Licensing OK?
+    # - intel-oneapi-compilers@2022.1
+    # - nvhpc
+
+  - cuda_specs:
+# Disabled for consistency with aarch64
+    #- relion +cuda cuda_arch=70
+    - raja +cuda cuda_arch=70
+    - mfem +cuda cuda_arch=70
+
+  - app_specs: 
+    - bwa
+# Disabled for consistency with aarch64
+    #- bowtie2
+# Disabled for consistency with aarch64
+    #- cistem
+    - cromwell
+    - fastqc
+    - flux-sched
+    - flux-core
+    - flux-pmix
+    - gatk
+    - gromacs
+    - lammps
+    - wrf build_type=dm+sm
+    - mfem
+    - mpas-model ^parallelio+pnetcdf
+    - nextflow
+    - octave
+    - openfoam
+    - osu-micro-benchmarks
+    - parallel
+    - paraview
+    - picard
+    - quantum-espresso
+    - raja
+# Disabled for consistency with aarch64
+    #- rsem
+    # Errors on texlive
+    #- rstudio
+    - salmon
+    - samtools
+    - seqtk
+    - snakemake
+    - star
+    # Requires gcc@9:
+    #- ufs-weather-model
+# Disabled for consistency with aarch64
+    #- visit
+    
+  - lib_specs:
+    - openmpi fabrics=ofi
+    - openmpi fabrics=ofi +legacylaunchers
+    - openmpi fabrics=auto
+    - mpich
+    - libfabric
+
+  - compiler:
+    - '%gcc'
```

Comment:
Why no compiler version?  I guess I don't see why it would be a problem off the top of my head, but I'm wondering if some bits in the pipeline generation would have an issue with it.

Reply:
Originally I was trying to get gitlab to run multiple images (ubuntu1804 and alinux2), and the versions of gcc are different (gcc@7.5.0 and gcc@7.3.1). Since multi-image does not seem to be supported within the spack.yaml, I think it does make sense to call out the compiler version. Also, as we get more compilers built in the build cache, I plan to expand the compiler block to include multiple families and versions (gcc, nvhpc, aocc, etc.).